tests/test_advection.py

```diff
@@ -416,13 +425,14 @@ def test_peninsula_fieldset(kernel, rtol, grid_type):
     start_lat = np.linspace(3e3, 47e3, npart)
     start_lon = 3e3 * np.ones_like(start_lat)
 
-    if kernel == AdvectionRK45:
-        fieldset.add_constant("RK45_tol", rtol)
-
     SampleParticle = Particle.add_variable(
         [Variable("p", initial=0.0, dtype=np.float32), Variable("p_start", initial=0.0, dtype=np.float32)]
     )
 
+    if kernel == AdvectionRK45:
+        fieldset.add_constant("RK45_tol", rtol)
+        SampleParticle = SampleParticle.add_variable(Variable("next_dt"))
```

Comment:
as above

Reply:
Implemented in ef12d26119a0babe61f61f005587fca641c39a3f